pkg/isoeditor/stream.go

```diff
@@ -29,7 +47,69 @@ func NewRHCOSStreamReader(isoPath string, ignitionContent *IgnitionContent, ramd
 		return nil, err
 	}
 
-	r, err := readerForFileContent(isoPath, ignitionImagePath, isoReader, ignitionReader)
+	// Starting with version 0.17.0 of the CoreOS installer a new '/coreos/igninfo.json' file may
+	// exist to indicate the location of the ignition inside the ISO. For example, in the S390
+	// platform it will contain something like this:
+	//
+	//	{
+	//		"file": "images/cdboot.img",
+	//		"offset": 66497660,
+	//		"length": 262144
+	//	}
+	//
+	// If it exists we need to read it, otherwise we can use the '/images/ignition.img' location
+	// used in previous versions.
+	var ignInfo IgnInfo
+	ignInfoBytes, err := ReadFileFromISO(isoPath, ignitionInfoPath)
+	if err != nil {
+		log.WithError(err).WithFields(log.Fields{
```

Comment:
I'd say this should not be logged as an error, as we know this is done for backward compatibility.
Maybe change it to warning?

Reply:
Is not logged as an error, but as `INFO`. The `WithError` call only means "add this error as a parameter". The resulting message is something like this:

```
time="2023-10-30T08:25:40+01:00" level=info msg="Failed to read ignition information file, will assume that it doesn't exist and use the default ignition location" default=/images/ignition.img error="could not read directory entries for /coreos" file=/coreos/igninfo.json iso=/tmp/1408693960test.iso
```

I can change it to warning if you prefer that.